### ISSUE TITLE
PWAs crash while 'Open links in incognito tabs always' is enabled

### ISSUE THREAD

**Issue by Eddi2015** [bromite/bromite#745]
### Bromite version

Version: `85.0.4183.110` (latest)
Arch: `arm64`
Android version: `9.0`, `10.0`

### Is this bug about the SystemWebView?

No.

### Is the bug reproducible with latest version?

Yes.

### Can the bug be reproduced with corresponding Chromium version?

Always icognito tab feature does not exist upstream, thus impossible to reproduce.

### Is the bug a crash?

yes, I did not have time yet to record a logcat. Maybe you can reproduce the issue yourself.

### Describe the bug

When trying to open a shortcut to a PWA, it crashes while 'Open links in incognito tabs always' is enabled. When disabling that feature the bug does not happen and PWAs open without a crash. The issue does not seem to happen while Bromite is running in the background. (reminds me of the LocationBarCrash which also only happened on 'cold' starts)

### Steps to reproduce the bug

Steps to reproduce the bug:
0. if already enabled, then disable the 'Open links in incognito tabs always' option in Bromite's privacy menu for now
2. Add a PWA shortcut (you can't add home screen shortcuts in incognito mode)
   example PWA list:
   -> https://tuxurls.com/
   -> https://geizhals.de/
   -> https://www.spiegel.de/ (put this site on javascript blacklist in order to browse it)
3. now enable  'Open links in incognito tabs always' and restart bromite
4. exit bromite
5. open a PWA while bromite is not running in the background

### Expected behavior

no crash; normal behaviour as if 'always incognito tabs' is disabled

**Comment by uazo** [comment]
crash stack

```
09-19 08:16:22.606 12943 12943 W System.err: java.lang.IllegalStateException: Attempting to open tab in wrong model
09-19 08:16:22.615 12943 12943 W System.err:    at org.chromium.chrome.browser.tabmodel.TabModelImpl.addTab(TabModelImpl.java:182)
09-19 08:16:22.615 12943 12943 W System.err:    at org.chromium.chrome.browser.customtabs.content.CustomTabActivityTabController.finalizeCreatingTab(CustomTabActivityTabController.java:335)
09-19 08:16:22.615 12943 12943 W System.err:    at org.chromium.chrome.browser.customtabs.content.CustomTabActivityTabController.finishNativeInitialization(CustomTabActivityTabController.java:241)
09-19 08:16:22.615 12943 12943 W System.err:    at org.chromium.chrome.browser.customtabs.BaseCustomTabActivity.initializeState(BaseCustomTabActivity.java:331)
09-19 08:16:22.615 12943 12943 W System.err:    at org.chromium.chrome.browser.init.ChromeBrowserInitializer.lambda$handlePostNativeStartup$5$ChromeBrowserInitializer(ChromeBrowserInitializer.java:290)
09-19 08:16:22.616 12943 12943 W System.err:    at org.chromium.chrome.browser.init.ChromeBrowserInitializer$$Lambda$6.run(Unknown Source:2)
09-19 08:16:22.616 12943 12943 W System.err:    at org.chromium.chrome.browser.init.ChainedTasks$1.run(ChainedTasks.java:33)
09-19 08:16:22.616 12943 12943 W System.err:    at android.os.MessageQueue.nativePollOnce(Native Method)
09-19 08:16:22.616 12943 12943 W System.err:    at android.os.MessageQueue.next(MessageQueue.java:326)
09-19 08:16:22.616 12943 12943 W System.err:    at android.os.Looper.loop(Looper.java:160)
09-19 08:16:22.616 12943 12943 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6718)
09-19 08:16:22.616 12943 12943 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
09-19 08:16:22.616 12943 12943 W System.err:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
09-19 08:16:22.616 12943 12943 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

same bug as https://github.com/bromite/bromite/issues/453, we need to properly initialize the webapp.

here a working patch:

```
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappIntentDataProvider.java b/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappIntentDataProvider.java
index fefb09b4c7..0a5e1e4a2d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappIntentDataProvider.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappIntentDataProvider.java
@@ -19,6 +19,9 @@ import org.chromium.chrome.browser.browserservices.BrowserServicesIntentDataProv
 import org.chromium.chrome.browser.flags.ActivityType;
 import org.chromium.components.browser_ui.widget.TintedDrawable;

+import org.chromium.base.ContextUtils;
+import org.chromium.chrome.browser.AlwaysIncognitoLinkInterceptor;
+
 /**
  * Stores info about a web app.
  */
@@ -32,6 +35,8 @@ public class WebappIntentDataProvider extends BrowserServicesIntentDataProvider
     private final @ActivityType int mActivityType;
     private final Intent mIntent;

+    private boolean mIsIncognito = false;
+
     /**
      * Returns the toolbar color to use if a custom color is not specified by the webapp.
      */
@@ -51,6 +56,10 @@ public class WebappIntentDataProvider extends BrowserServicesIntentDataProvider
         mWebappExtras = webappExtras;
         mWebApkExtras = webApkExtras;
         mActivityType = (webApkExtras != null) ? ActivityType.WEB_APK : ActivityType.WEBAPP;
+
+        if (ContextUtils.getAppSharedPreferences().getBoolean(AlwaysIncognitoLinkInterceptor.PREF_ALWAYS_INCOGNITO, false)) {
+            mIsIncognito = true;
+        }
     }

     @Override
@@ -128,4 +137,9 @@ public class WebappIntentDataProvider extends BrowserServicesIntentDataProvider
     public WebApkExtras getWebApkExtras() {
         return mWebApkExtras;
     }
+
+    @Override
+    public boolean isIncognito() {
+        return mIsIncognito;
+    }
```

by doing so, the webapps are launched in incognito mode if the flag is active. @csagan5 Is that correct? No data would be kept between sessions.

**Comment by csagan5** [comment]
@uazo sure, shall I update the patch or you prefer to submit a PR?

**Comment by uazo** [comment]
sorry, sure, if you can it's better for me now

**Comment by csagan5** [comment]
Fixed in `86.0.4240.66`.

**Comment by kokukyocho** [comment]
Hello @Eddi2015!

I am a bot :robot:

Thanks for submitting this issue!
I noticed that it is missing the template, please edit the issue to match the template for either a [bug](https://github.com/bromite/bromite/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) or a [feature request](https://github.com/bromite/bromite/blob/master/.github/ISSUE_TEMPLATE/feature_request.md).

The issue will be automatically re-opened after that.

**Tip**: if you use GitHub in Desktop mode the template will be proposed to you when submitting issues.
